tools/inkscape_extension/visicut_export.py

```diff
@@ -326,6 +333,17 @@ def get_original_filename(filename):
         VISICUTBIN = which("VisiCut.Linux", [VISICUTDIR, "/usr/share/visicut"])
 INKSCAPEBIN = which("inkscape", [INKSCAPEDIR])
 
+## Test if this inkscape is in an appimage.
+# We detect this by checking for an AppRun file, in one of the parent folders of our INKSCAPEBIN.
+# If so, replace INKSCAPEBIN with AppRun, as this is the only safe way to call inkscape.
+# (a direct call mixes libraries from the host system with the appimage, may or may not work.)
+dir = os.path.split(INKSCAPEBIN)[0]
+while dir != '/':
+    if os.path.exists(os.path.join(dir, "AppRun")):
+        INKSCAPEBIN = os.path.join(dir, "AppRun")
+        break
+    dir = os.path.split(dir)[0]
+
```

Comment:
```suggestion
dir = os.path.dirname(INKSCAPEBIN)
while dir != '/':
    apprun_path = os.path.join(dir, "AppRun")
    if os.path.exists(apprun_path):
        INKSCAPEBIN = apprun_path
        break
    dir = os.path.dirname(dir)
```

What is the compatibility target for this code? Python 3.4 introduced `pathlib` which makes such code a bit more convenient.

Reply:
Please also bear in mind that this code will make the extension work *only* for the currently running Inkscape. The following approach should make it work permanently:

```
appimage = os.environ.get("APPIMAGE")
if appimage and os.path.exists(appimage):
    INKSCAPE_BIN = appimage
else:
    # code from above
```

The reason to include the other code is that the AppImage _might_ be extracted by some users to a static location. Yes, that's very much annoying and yes, they're technically not AppImages any more. But that's what some people do. Personally, I don't care as much, I'm happy with the three-liner already.

Reply:
In principal, I'd be happy to keep compatibility with all python versions, currently supported. 
As we support a very broad range of platforms, I don't want to make assumptions what python might be running in all our linux, windows or mac environments.

Apparently we already use array unpacking in `subprocess.Popen(
            [VISICUTBIN, *arguments, dest_filename]` which was introduced in python 3.5(?) :see_no_evil: 
This probably already limits backwards compatibility in an unintended way.

Reply:
Then again, Python 3.4 and 3.5 were released in 2014 and 2015 respectively hand have come EOL in 2019 and 2020 respectively. There's no point in keeping compatibility to dead software.

The oldest distro that might still use Python 3.4 is CentOS 7 which is due for EOL in June, i.e., not even 2 months from now.

Reply:
Thus, the compatibility target question is answerd :-)

Reply:
Afaik, the environment variable APPIMAGE points to the appimage file, rather than to anything inside a mounted or unpacked appimage.
I've chosen the latter, as this is noticeably more efficient.

I don't understand your comment about "the extension work only for the currently running Inkscape" -- isn't that the whole point of an extension to run with the currently running inkscape? 
I cant imagine a real world scenario, where the original inkscape was closed before the extension would make its cleanup call. 
(That might exist as a race condition, if the user starts the extension and then quickly kills inkscape. But what should the user expect in that case anyway?)